default/python/AI/ProductionAI.py

```diff
@@ -97,10 +97,15 @@ def cur_best_military_design_rating():
         return _best_military_design_rating_cache[current_turn]
     priority = PriorityType.PRODUCTION_MILITARY
     if priority in _design_cache:
+        # TODO: rework the below code, CombatRatingsAI and possibly  WarShipDesigner so that the below uses a rating
+        # provided by CombatRatingsAI (so as to be more suitable for its actual use and to avoid making any assumptions
+        # about how the WarShipDesigner may apply cost adjustments to its rating).
```

Comment:
I agree. This is definitely the wrong place to calculate the rating "backwards".

As far as I can see, the rating you calculate here may not even be correct because of the ```_adjusted_production_cost``` used when calculating the rating:
 https://github.com/freeorion/freeorion/blob/master/default/python/AI/ShipDesignAI.py#L1457
and also some of the non-combat-stuff (fuel, speed) that are not relevant to the rating as interpreted by ```cur_best_military_design_rating()```

In my opionion, the cleanest approach is to give the ```WarShipDesigner``` an interface to calculate and return a combat rating based on the calculations in ```CombatRatingsAI```.
That rating may or may not be used within the ```rating_function``` but is not equivalent to it (as it neglects e.g. fuel)